Add a single source of build and release configuration (#42000)
rancher/rancher: release.md

Reviewer: Please edit or delete `docs/release.md` if this will be the new location.

Author: I think we could delete it anyways, since we haven't been using it, as far as I understand. Release captains use instructions found in Confluence...

## One place for release configuration

Release captains should edit a single file to update any configuration values.

> Those values typically need to have set versions before each release.

Reviewer 2: ```suggestion
Rancher relies on various sub-components, such as the webhook. These typically need to have set versions for Rancher to build properly.
```

> They are found in different places, like Dockerfiles, Go files, etc.

Reviewer 2: ```suggestion
Build variables can be used in a variety of places/ways, including as Env Vars in Dockerfiles, and Constants in go code. 
```

The `config.yaml` is the single source of truth. It lists all values by name and value.
Changes to it should be committed to source control.

### Update an existing value

Edit the config.yaml file and update the desired value.

> Then run any relevant scripts.

Reviewer 2: ```suggestion
Next, re-build Rancher using `make build`, or re-package rancher using `make package`.
```

### Add a new value

Add it to config.yaml.

> Then update the `config` script. For example:

Reviewer 2: Would also be good to include a general guide on updating the places that it is used - the below will create a new build var but it won't actually get used when building rancher.
